### DISCUSSION TITLE
Removing a provider station while it's on schedule

### DISCUSSION THREAD

**Discussion by Peekayy** [Opinionated-Blueprints/10-Books-Full-of-Rails#33] [Q&A] [answered]
First i'm new to this provider/requester stations thing but it looks very promissing !
I had an issue when I removed a provider station while one train was heading for it. Then all trains in the group (including new trains from blueprint) were trying to reach the deleted station. I fixed it by removing all trains of the group and starting over.

What I remember doing precisely :
- Notice train going to an unfinished stone provider station (this was the only stone station)
- Remove a rail in the station entrance so the train cannot reach it.
- Train goes to depot but still wants to go to stone while sulfur station is ready to accept a train.
- Remove the stone station
- Train still wants to go there, new trains too.

Did I miss something or is it some sort of caveat that shouldn't happen as soon as there are more than 1 station of a kind ?

Also : thanks for the great work.

**Comment by Bocian-1** [comment, marked as answer]
I don't know how that's possible honestly. It should just get 'train stop does not exist', get interrupted by depot, clear the stone station interrupt and reacquire target. And new trains? That makes no sense to me. It could be a bug in the game at this point.

**Comment by Bocian-1** [comment]
Also, for the future, something like this should be reported as an [issue](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/issues)

**Reply by Peekayy** [reply]
Dang sorry, I was pretty sure this issue wasn't related to the blueprint itself so I chose to put it here 😅
